lib/elixir/test/elixir/module/checker_test.exs

```diff
@@ -11,437 +11,498 @@ defmodule Module.CheckerTest do
+          @compile {:no_warn_undefined, {B, :func, 2}}
+
+          def a, do: MissingModule.func(1)
+          def b, do: MissingModule2.func(1, 2)
+          def c, do: MissingModule2.func(1)
+          def d, do: MissingModule3.func(1, 2)
+          def e, do: B.func(1)
+          def f, do: B.func(1, 2)
+          def g, do: B.func(1, 2, 3)
+        end
+        """,
+        "b.ex" => """
+        defmodule B do
+          def func(_), do: :ok
+        end
+        """
+      }
 
-    assert_warnings(files, warning)
-  end
+      warning = """
+      warning: MissingModule2.func/1 is undefined (module MissingModule2 is not available or is yet to be defined)
+        a.ex:7: A.c/0
 
-  test "imports" do
-    files = %{
-      "a.ex" => """
-      defmodule A do
-        import Record
+      warning: MissingModule3.func/2 is undefined (module MissingModule3 is not available or is yet to be defined)
+        a.ex:8: A.d/0
 
-        def a(a, b), do: extract(a, b)
-        def b(arg), do: is_record(arg)
-      end
-      """
-    }
+      warning: B.func/3 is undefined or private. Did you mean one of:
 
-    assert_no_warnings(files)
-  end
+            * func/1
 
-  test "requires" do
-    files = %{
-      "a.ex" => """
-      defmodule A do
-        require Integer
+        a.ex:11: A.g/0
 
-        def a(a), do: Integer.is_even(a)
-      end
       """
-    }
 
-    assert_no_warnings(files)
+      assert_warnings(files, warning)
+    end
   end
 
-  test "excludes no_warn_undefined" do
-    files = %{
-      "a.ex" => """
-      defmodule A do
-        @compile {:no_warn_undefined, [MissingModule, {MissingModule2, :func, 2}]}
-        @compile {:no_warn_undefined, {B, :func, 2}}
-
-        def a, do: MissingModule.func(1)
-        def b, do: MissingModule2.func(1, 2)
-        def c, do: MissingModule2.func(1)
-        def d, do: MissingModule3.func(1, 2)
-        def e, do: B.func(1)
-        def f, do: B.func(1, 2)
-        def g, do: B.func(1, 2, 3)
-      end
-      """,
-      "b.ex" => """
-      defmodule B do
-        def func(_), do: :ok
-      end
-      """
-    }
+  describe "deprecated" do
+    test "reports deprecated functions" do
+      files = %{
+        "a.ex" => """
+        defmodule A do
+          @deprecated "oops"
+          def a, do: A.a
+        end
+        """
+      }
 
-    warning = """
-    warning: MissingModule2.func/1 is undefined (module MissingModule2 is not available or is yet to be defined)
-      a.ex:7: A.c/0
+      warning = """
+      warning: A.a/0 is deprecated. oops
+        a.ex:3: A.a/0
 
-    warning: MissingModule3.func/2 is undefined (module MissingModule3 is not available or is yet to be defined)
-      a.ex:8: A.d/0
+      """
 
-    warning: B.func/3 is undefined or private. Did you mean one of:
+      assert_warnings(files, warning)
+    end
 
-          * func/1
+    # test "reports deprecated structs" do
```

Comment:
Unfortunately I don't think we can make this work. Before we tracked all dependencies but since the group pass only looks at runtime code we have lost this information.